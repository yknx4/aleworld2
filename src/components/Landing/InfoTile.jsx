import React,{Component, PropTypes} from 'react';
import {ImageTile} from 'components/layout';
import ReactMarkdown from 'react-markdown';

export default class InfoTile extends Component {

    render() {
      const {description, picture, subtitle, title} = this.props;
      const {src, alt} = picture;
        return (
          <ImageTile className="outer-title text-justify">
            <img alt={alt} src={src} className="img-circle sm-img"/>
            <div className="title mb16">
              <h5 className="uppercase mb0">{title}</h5>
              <span>{subtitle}</span>
            </div>
            <ReactMarkdown
              source={description}
              className="mb0"
              softBreak="br"/>
          </ImageTile>
        );
    }
}

InfoTile.propTypes = {
  description: PropTypes.string.isRequired,
  picture: PropTypes.shape({
    alt: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  }),
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
};
