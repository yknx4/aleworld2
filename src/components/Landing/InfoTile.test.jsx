import React from 'react';
import { shallow, mount } from 'enzyme';
import InfoTile from './InfoTile';

describe('InfoTile', () => {
  let shallowComponent;
  let mountedComponent;

  const props = {
    description: 'description',
    title: 'title',
    subtitle: 'subtitle',
    picture: {
      src: '//path',
      alt: 'alt'
    }
  };

  beforeEach(() => {
    const component = <InfoTile {...props} > <p /> </InfoTile>;
    shallowComponent = shallow(component);
    mountedComponent = mount(component);
  });

  it('should render without throwing an error', function() {
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should be selectable by tag', function() {
    expect(shallowComponent.is('ImageTile')).toBe(true);
    expect(shallowComponent).toMatchSnapshot();
  });

  it('should mount in a full DOM', function() {
    expect(mountedComponent.find('ImageTile').length).toBe(1);
    expect(mountedComponent).toMatchSnapshot();
  });

});
