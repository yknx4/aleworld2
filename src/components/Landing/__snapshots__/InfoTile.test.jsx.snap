// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTile should be selectable by tag 1`] = `
<ImageTile
  className="outer-title text-justify"
>
  <img
    alt="alt"
    className="img-circle sm-img"
    src="//path"
  />
  <div
    className="title mb16"
  >
    <h5
      className="uppercase mb0"
    >
      title
    </h5>
    <span>
      subtitle
    </span>
  </div>
  <ReactMarkdown
    className="mb0"
    containerTagName="div"
    parserOptions={Object {}}
    softBreak="br"
    source="description"
  />
</ImageTile>
`;

exports[`InfoTile should mount in a full DOM 1`] = `
<InfoTile
  description="description"
  picture={
    Object {
      "alt": "alt",
      "src": "//path",
    }
  }
  subtitle="subtitle"
  title="title"
>
  <ImageTile
    className="outer-title text-justify"
  >
    <Module
      baseClass="image-tile"
      className="outer-title text-justify"
      tag="div"
    >
      <div
        className="image-tile outer-title text-justify"
      >
        <img
          alt="alt"
          className="img-circle sm-img"
          src="//path"
        />
        <div
          className="title mb16"
        >
          <h5
            className="uppercase mb0"
          >
            title
          </h5>
          <span>
            subtitle
          </span>
        </div>
        <ReactMarkdown
          className="mb0"
          containerTagName="div"
          parserOptions={Object {}}
          softBreak="br"
          source="description"
        >
          <div
            className="mb0"
          >
            <p>
              description
            </p>
          </div>
        </ReactMarkdown>
      </div>
    </Module>
  </ImageTile>
</InfoTile>
`;

exports[`InfoTile should render without throwing an error 1`] = `
<ImageTile
  className="outer-title text-justify"
>
  <img
    alt="alt"
    className="img-circle sm-img"
    src="//path"
  />
  <div
    className="title mb16"
  >
    <h5
      className="uppercase mb0"
    >
      title
    </h5>
    <span>
      subtitle
    </span>
  </div>
  <ReactMarkdown
    className="mb0"
    containerTagName="div"
    parserOptions={Object {}}
    softBreak="br"
    source="description"
  />
</ImageTile>
`;
